### PR TITLE
sql,server: add VIEWACTIVITYREDACTED role

### LINKED ISSUE #74716: Create a new role option (VIEWACTIVITIYREDACTED) which will not have access to PII within the console
This issue tracks introducing a new role option (VIEWACTIVITYREDACTED) which will hide and disable statement bundles access in the console. Specifically,

- Statements overview page - Diagnostics column
- Statements Details - Diagnostics tab
- Advanced Debug - Statement diagnostics history

This role option will give the user the exact same access as VIEWACTIVITY but will make PII information inaccessible such as statement bundles.

Epic: CRDB-12103

### PR DESCRIPTION
**Pull request by maryliag**
This commit adds the new `VIEWACTIVITYREDACTED` role. This role
should act the same as `VIEWACTIVITY` for the majority of uses,
but it restricts the usage of Statements Diagnostics Bundle, which
contains PII information.
It is possible for a user to have both `VIEWACTIVITY` and
`VIEWACTIVITYREDACTED`, but in this case the `VIEWACTIVITYREDACTED`
takes precedent and the user can't use the features restricted by it.

Fixes #74716

Release note (sql change): Creation of the new role `VIEWACTIVITYREDACTED`
that works similar as `VIEWACTIVITY` but restricts the usage of
Statements Diagnostics Bundle. It is possible for a user to have both roles
(`VIEWACTIVITY` and `VIEWACTIVITYREDACTED`), but the role `VIEWACTIVITYREDACTED`
takes precedent on restrictions.

### CODE REVIEW

**Review by Azhng — COMMENTED** [review]
We should also probably introduce some tests here to see if this new option is working as intended. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-:-MtAUoj83lnqLr0ra9xr:bsjumi)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian, @maryliag, and @RichardJCai)

___
*[pkg/server/admin.go, line 3113 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtATu4JArW6jUHd-yen:-MtATu4JArW6jUHd-yeo:b7d2rua) ([raw file](https://github.com/cockroachdb/cockroach/blob/e69b9082cca93f3f474c9d415550a9ee8d6e9805/pkg/server/admin.go#L3113)):*
> ```Go
> 			}
> 
> 			if !hasViewActivity {
> ```

Hmm, adding boolean flags here is making the switch logic here a bit convoluted. How about just defining a separate function to perform the check on `roleoption.VIEWACTIVITYREDACTED`? 

I'm thinking something along the line of:


```golang

// replaces the current requireViewActivityPermission function
func (c *adminPrivilegeChecker) requireViewActivityOrViewAcitivtyRedactedPermission() {}

// this functions fails if user only have the redacted permission, and only used in the statement bundles? 
func (c *adminPrivilegeChecker) requireViewActivityPermission() {}

```

**Review by maryliag — COMMENTED** [review]
I'll be adding tests, I just wanted to open the PR for initial reviews of the implementation :) 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-:-MtAjqdQ3u8VkokeQmf_:bxrrs0f)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng, @dhartunian, and @RichardJCai)

___
*[pkg/server/admin.go, line 3113 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtATu4JArW6jUHd-yen:-MtAivgc0zUResV_uFjP:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/e69b9082cca93f3f474c9d415550a9ee8d6e9805/pkg/server/admin.go#L3113)):*
<details><summary><i>Previously, Azhng (Archer Zhang) wrote…</i></summary><blockquote>

Hmm, adding boolean flags here is making the switch logic here a bit convoluted. How about just defining a separate function to perform the check on `roleoption.VIEWACTIVITYREDACTED`? 

I'm thinking something along the line of:


```golang

// replaces the current requireViewActivityPermission function
func (c *adminPrivilegeChecker) requireViewActivityOrViewAcitivtyRedactedPermission() {}

// this functions fails if user only have the redacted permission, and only used in the statement bundles? 
func (c *adminPrivilegeChecker) requireViewActivityPermission() {}

```
</blockquote></details>

Done
___
*[pkg/sql/sem/builtins/builtins.go, line 5412 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtAjVKNANLLf42P8O-N:-MtAjVKNANLLf42P8O-O:b-rr6kxl) ([raw file](https://github.com/cockroachdb/cockroach/blob/2b77291db0a178f6c17c9f4f5f73747c59c50aaf/pkg/sql/sem/builtins/builtins.go#L5412)):*
> ```Go
> 					return nil, errors.AssertionFailedf("session accessor not set")
> 				}
> 				optionStr := string(tree.MustBeDString(args[0]))
> ```

on this step the new role `VIEWACTIVITYREDACTED` is being passed as `VIEWACTIVITYRED` on args and I don't understand why the value is being cut off (which is the reason why this is failing tests)

**Review by matthewtodd — COMMENTED** [review]
> Reviewed 4 of 10 files at r1, 6 of 8 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74715)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng, @dhartunian, @maryliag, and @RichardJCai)

___
*[pkg/server/admin.go, line 3133 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtDhQFfEHSBPTi-nuRf:-MtDhQFgE6qLq74awHEa:bq30xnm) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f4354a1fc52ac9f29a0ebfffc0817c1d0366b4e/pkg/server/admin.go#L3133)):*
> ```Go
> 			if !hasViewActivity {
> 				return userName, status.Errorf(
> 					codes.PermissionDenied, "this operation requires the %s role option",
> ```

Would it make sense to also mention `VIEWACTIVITYREDACTED` in this error message?
___
*[pkg/sql/sem/builtins/builtins.go, line 5412 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtAjVKNANLLf42P8O-N:-MtDiSx7-eAvGpXfe4z6:b-pkcwqt) ([raw file](https://github.com/cockroachdb/cockroach/blob/2b77291db0a178f6c17c9f4f5f73747c59c50aaf/pkg/sql/sem/builtins/builtins.go#L5412)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

on this step the new role `VIEWACTIVITYREDACTED` is being passed as `VIEWACTIVITYRED` on args and I don't understand why the value is being cut off (which is the reason why this is failing tests)
</blockquote></details>

Neat, 15 feels like 16 minus a null terminator.
___
*[pkg/sql/sem/builtins/builtins.go, line 5396 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtDiBEJBOSdPMrKMIRD:-MtDiBEJBOSdPMrKMIRE:b5pgn21) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/sql/sem/builtins/builtins.go#L5396)):*
> ```Go
> 	),
> 
> 	// Returns true iff the current user has the specified role option.
> ```

I imagine this "iff" intentionally meant "[if and only if](https://en.wikipedia.org/wiki/If_and_only_if)."

**Review by maryliag — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74715)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng, @dhartunian, @matthewtodd, and @RichardJCai)

___
*[pkg/server/admin.go, line 3133 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtDhQFfEHSBPTi-nuRf:-MtDl0yj-rDI9TESi7Gk:b-dw7ral) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f4354a1fc52ac9f29a0ebfffc0817c1d0366b4e/pkg/server/admin.go#L3133)):*
<details><summary><i>Previously, matthewtodd (Matthew Todd) wrote…</i></summary><blockquote>

Would it make sense to also mention `VIEWACTIVITYREDACTED` in this error message?
</blockquote></details>

Good point! Done
___
*[pkg/sql/sem/builtins/builtins.go, line 5412 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtAjVKNANLLf42P8O-N:-MtDnac34-B7FRcNWWmH:b38iuzv) ([raw file](https://github.com/cockroachdb/cockroach/blob/2b77291db0a178f6c17c9f4f5f73747c59c50aaf/pkg/sql/sem/builtins/builtins.go#L5412)):*
<details><summary><i>Previously, matthewtodd (Matthew Todd) wrote…</i></summary><blockquote>

Neat, 15 feels like 16 minus a null terminator.
</blockquote></details>

Found my error: I was using a different role name initially, after I changed to VIEWACTIVITYREDACTED I forgot to update the index on https://github.com/cockroachdb/cockroach/pull/74715/files#diff-62f5aa0de7b141365466951226b37b27853245ce378400965430a9adc38b5867R38



_Code quote (from docs/generated/sql/bnf/stmt_block.bnf):_
```
MODIFYCLUSTERSETTING
```
___
*[pkg/sql/sem/builtins/builtins.go, line 5396 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtDiBEJBOSdPMrKMIRD:-MtDlDgv-C5l6KYVRbBX:b2nhoia) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/sql/sem/builtins/builtins.go#L5396)):*
<details><summary><i>Previously, matthewtodd (Matthew Todd) wrote…</i></summary><blockquote>

I imagine this "iff" intentionally meant "[if and only if](https://en.wikipedia.org/wiki/If_and_only_if)."
</blockquote></details>

TIL



_Code quote:_
```Go
Returns true if the current user has the specifi
```

File: pkg/server/admin.go [hunk 1]
```diff
@@ -3108,6 +3109,35 @@ func (c *adminPrivilegeChecker) requireViewActivityPermission(
...
+		return userName, err
+	}
+	if !isAdmin {
+		hasViewActivity, err := c.hasRoleOption(ctx, userName, roleoption.VIEWACTIVITY)
+		if err != nil {
+			return userName, err
+		}
+
+		if !hasViewActivity {
+			hasViewActivity, err = c.hasRoleOption(ctx, userName, roleoption.VIEWACTIVITYREDACTED)
```
Comment (RichardJCai) [inline pkg/server/admin.go:3126]: nit: call this `hasViewActivityRedacted`

**Review by RichardJCai — COMMENTED** [review]
Implementation looks good so far, will approve after tests are added

**Review by dhartunian — COMMENTED** [review]
> Reviewed 1 of 10 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74715)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng, @dhartunian, @maryliag, @matthewtodd, and @RichardJCai)

___
*[pkg/server/status.go, line 166 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtE6Olp4yym89lsn-_X:-MtE6Olp4yym89lsn-_Y:b-qn0atn) ([raw file](https://github.com/cockroachdb/cockroach/blob/4bbb53adbe0f142e1677d566d7e9eb24ba309cdd/pkg/server/status.go#L166)):*
> ```Go
> 	}
> 	if !hasViewActivity {
> 		hasViewActivity, err = b.privilegeChecker.hasRoleOption(ctx, sessionUser, roleoption.VIEWACTIVITYREDACTED)
> ```

can this and the code above use `hasViewActivityOrViewActivityRedacted()` defined below?

**Review by maryliag — COMMENTED** [review]
Tests created

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-:-MtFR7HL8ayxDEoT_y8h:blypcro)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng, @dhartunian, @matthewtodd, and @RichardJCai)

___
*[pkg/server/admin.go, line 3126 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtAOjmPDRoAIqcA1_j_-r4-3126:-MtEnjb88IdLg8h7gKfF:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/4bbb53adbe0f142e1677d566d7e9eb24ba309cdd/pkg/server/admin.go#L3126)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

nit: call this `hasViewActivityRedacted`
</blockquote></details>

Done
___
*[pkg/server/status.go, line 166 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtE6Olp4yym89lsn-_X:-MtEnmD--MA82CWyT7Ps:b499q7j) ([raw file](https://github.com/cockroachdb/cockroach/blob/4bbb53adbe0f142e1677d566d7e9eb24ba309cdd/pkg/server/status.go#L166)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

can this and the code above use `hasViewActivityOrViewActivityRedacted()` defined below?
</blockquote></details>

Changed

**Review by RichardJCai — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74715)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng, @dhartunian, @maryliag, @matthewtodd, and @RichardJCai)

___
*[pkg/sql/crdb_internal.go, line 956 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtKYxPwDOuOUklw7PHu:-MtKYxPwDOuOUklw7PHv:b5src4s) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e1c0fbd4db9da222b59b294ec1c83f72088febb/pkg/sql/crdb_internal.go#L956)):*
> ```Go
> )`,
> 	populate: func(ctx context.Context, p *planner, _ catalog.DatabaseDescriptor, addRow func(...tree.Datum) error) error {
> 		hasViewActivity, err := p.HasRoleOption(ctx, roleoption.VIEWACTIVITY)
> ```

Can we factor out this call here? We duplicate it below but also somewhat duplicate this logic in `requireViewActivityOrViewActivityRedactedPermission` and `hasViewActivityOrViewActivityRedactedPermissions` can we have one helper function that checks for `hasViewActivity` and `hasViewActivityRedacted` and use that in all these places?
___
*[pkg/server/status_test.go, line 1912 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtKZiXkDxGCDxloU5ZA:-MtKZiXl9qko2b8l-YQJ:b-c32eec) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e1c0fbd4db9da222b59b294ec1c83f72088febb/pkg/server/status_test.go#L1912)):*
> ```Go
> 
> 	// Grant VIEWACTIVITYREDACTED.
> 	thirdServerSQL.Exec(t, fmt.Sprintf("ALTER USER %s VIEWACTIVITYREDACTED", authenticatedUserNameNoAdmin().Normalized()))
> ```

Do we have to revoke `VIEWACTIVITY` before doing this? Otherwise the user has both?
I guess another question is does it make sense for a user to have both `VIEWACTIVITY` and `VIEWACTIVITYREDACTED`? What should happen there?
___
*[pkg/server/status_test.go, line 2046 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtKZxbrBtyEmCAqT96p:-MtKZxbrBtyEmCAqT96q:b-d4zb0i) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e1c0fbd4db9da222b59b294ec1c83f72088febb/pkg/server/status_test.go#L2046)):*
> ```Go
> 	testPath(fmt.Sprintf("combinedstmts?start=%d", oneMinAfterAggregatedTs), nil)
> 
> 	// Grant VIEWACTIVITYREDACTED.
> ```

Ditto with above

**Review by maryliag — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74715)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng, @dhartunian, @kevin-v-ngo, @maryliag, @matthewtodd, and @RichardJCai)

___
*[pkg/server/status_test.go, line 1912 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtKZiXkDxGCDxloU5ZA:-MtK_hmx9RgTxGNCu6-L:bx6avxy) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e1c0fbd4db9da222b59b294ec1c83f72088febb/pkg/server/status_test.go#L1912)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

Do we have to revoke `VIEWACTIVITY` before doing this? Otherwise the user has both?
I guess another question is does it make sense for a user to have both `VIEWACTIVITY` and `VIEWACTIVITYREDACTED`? What should happen there?
</blockquote></details>

For this test I need to revoke the VIEWACTIVITY, because that role gives permission for the diagnostics, but the REDACTED doesn't, which is what I'm trying to test here.
It is possible for the user to have both roles, in that case, if they have VIEWACTIVITY they can use the features normally.

@kevin-v-ngo should we remove VIEWACTIVITY if the user was assigned with VIEWACTIVITYREDACTED?

**Review by maryliag — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74715)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng, @dhartunian, @kevin-v-ngo, @maryliag, @matthewtodd, and @RichardJCai)

___
*[pkg/server/status_test.go, line 1912 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtKZiXkDxGCDxloU5ZA:-MtKksVs3FYIkcvkAsxS:bfxvm10) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e1c0fbd4db9da222b59b294ec1c83f72088febb/pkg/server/status_test.go#L1912)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

For this test I need to revoke the VIEWACTIVITY, because that role gives permission for the diagnostics, but the REDACTED doesn't, which is what I'm trying to test here.
It is possible for the user to have both roles, in that case, if they have VIEWACTIVITY they can use the features normally.

@kevin-v-ngo should we remove VIEWACTIVITY if the user was assigned with VIEWACTIVITYREDACTED?
</blockquote></details>

Talked with Kevin offline. It's possible for the user have both roles, but the REDACTED should take precedent, so I'll make the necessary changes here to take that into account

**Review by maryliag — COMMENTED** [review]
New changes implemented, to give precedent for VIEWACTIVITYREDACTED

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-:-MtL5A1wCbu_JUy7KiYg:b-ti4o0z)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng, @dhartunian, @kevin-v-ngo, @matthewtodd, and @RichardJCai)

___
*[pkg/sql/crdb_internal.go, line 956 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtKYxPwDOuOUklw7PHu:-MtL5Jpw0GSQffNq3KGL:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e1c0fbd4db9da222b59b294ec1c83f72088febb/pkg/sql/crdb_internal.go#L956)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

Can we factor out this call here? We duplicate it below but also somewhat duplicate this logic in `requireViewActivityOrViewActivityRedactedPermission` and `hasViewActivityOrViewActivityRedactedPermissions` can we have one helper function that checks for `hasViewActivity` and `hasViewActivityRedacted` and use that in all these places?
</blockquote></details>

Done
___
*[pkg/server/status_test.go, line 1912 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtKZiXkDxGCDxloU5ZA:-MtL5OHIDY5PiXDEU_M0:b-iaynra) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e1c0fbd4db9da222b59b294ec1c83f72088febb/pkg/server/status_test.go#L1912)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

Talked with Kevin offline. It's possible for the user have both roles, but the REDACTED should take precedent, so I'll make the necessary changes here to take that into account
</blockquote></details>

Realized that for this test I indeed should revoke the VIEWACTIVITY, since my goal here is to test if the VIEWACTIVITYREDACTED work the same way as the VIEWACTIVITY would. Made the change to remove the VIEWACTIVITY role here.
___
*[pkg/server/status_test.go, line 2046 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtKZxbrBtyEmCAqT96p:-MtL5fvu0T-mYwMM1zsI:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e1c0fbd4db9da222b59b294ec1c83f72088febb/pkg/server/status_test.go#L2046)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

Ditto with above
</blockquote></details>

done

**Review by RichardJCai — APPROVED** [review]
LGTM after last 3 comments are addressed

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-:-MtLErS9CUEeOG-wbjkE:b-t31o6x)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng, @dhartunian, @kevin-v-ngo, @maryliag, @matthewtodd, and @RichardJCai)

___
*[pkg/server/admin.go, line 3156 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtLEPNqA8aeK5lAjOSY:-MtLEPNqA8aeK5lAjOSZ:bmzkwde) ([raw file](https://github.com/cockroachdb/cockroach/blob/d32539601b06c6ab12dd6384517516850306a80a/pkg/server/admin.go#L3156)):*
> ```Go
> 		if hasViewActivityRedacted {
> 			return status.Errorf(
> 				codes.PermissionDenied, "this operation is not allowed for %s role option",
> ```

Should we also say they need `VIEWACTIVITY` in the error?
ie `VIEWACTIVITYREDACTED is insufficient...`
___
*[pkg/server/status_test.go, line 1911 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtLEZcN4vjoKglryIX8:-MtLEZcN4vjoKglryIX9:bd0ea06) ([raw file](https://github.com/cockroachdb/cockroach/blob/d32539601b06c6ab12dd6384517516850306a80a/pkg/server/status_test.go#L1911)):*
> ```Go
> 	testPath(fmt.Sprintf("statements?combined=true&start=%d", aggregatedTs+60), nil)
> 
> 	// Remove VIEWACTIVITY so we can test with just the VIEWACTIVITY role.
> ```

Replace the 2nd VIEWACTIVITY with VIEWACTIVITYREDACTED 
___
*[pkg/server/status_test.go, line 2048 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtLEgNx0_PE_-Qn_hk4:-MtLEgNx0_PE_-Qn_hk5:bd0ea06) ([raw file](https://github.com/cockroachdb/cockroach/blob/d32539601b06c6ab12dd6384517516850306a80a/pkg/server/status_test.go#L2048)):*
> ```Go
> 	testPath(fmt.Sprintf("combinedstmts?start=%d", oneMinAfterAggregatedTs), nil)
> 
> 	// Remove VIEWACTIVITY so we can test with just the VIEWACTIVITY role.
> ```

Replace the 2nd VIEWACTIVITY with VIEWACTIVITYREDACTED

**Review by maryliag — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74715)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Azhng, @dhartunian, @kevin-v-ngo, @matthewtodd, and @RichardJCai)

___
*[pkg/server/admin.go, line 3156 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtLEPNqA8aeK5lAjOSY:-MtNzbPJBOUgaESJw1gN:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/d32539601b06c6ab12dd6384517516850306a80a/pkg/server/admin.go#L3156)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

Should we also say they need `VIEWACTIVITY` in the error?
ie `VIEWACTIVITYREDACTED is insufficient...`
</blockquote></details>

Done
___
*[pkg/server/status_test.go, line 1911 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtLEZcN4vjoKglryIX8:-MtNzeybBw59f-heprZF:b-vauma2) ([raw file](https://github.com/cockroachdb/cockroach/blob/d32539601b06c6ab12dd6384517516850306a80a/pkg/server/status_test.go#L1911)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

Replace the 2nd VIEWACTIVITY with VIEWACTIVITYREDACTED 
</blockquote></details>

Done :facepalm: 
___
*[pkg/server/status_test.go, line 2048 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/74715#-MtLEgNx0_PE_-Qn_hk4:-MtNziH2CHrqzCZH2K3g:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/d32539601b06c6ab12dd6384517516850306a80a/pkg/server/status_test.go#L2048)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

Replace the 2nd VIEWACTIVITY with VIEWACTIVITYREDACTED 
</blockquote></details>

Done

**Comment by RichardJCai** [comment]
LGTM!

**Comment by maryliag** [comment]
TFTR
bors r+

**Comment by maryliag** [comment]
blathers backport 21.2

**Pull request merged**